mtags/src/main/scala-2/scala/meta/internal/pc/PcRenameProvider.scala

```diff
@@ -3,24 +3,50 @@ package scala.meta.internal.pc
 import scala.meta.internal.mtags.MtagsEnrichments._
 import scala.meta.pc.OffsetParams
 
-import org.eclipse.lsp4j.TextEdit
+import org.eclipse.{lsp4j => l}
 
 class PcRenameProvider(
     override val compiler: MetalsGlobal,
     params: OffsetParams,
-    name: String
-) extends PcCollector[TextEdit](compiler, params) {
+    name: Option[String]
+) extends PcCollector[l.TextEdit](compiler, params) {
   import compiler._
-  val newName: String = Identifier.backtickWrap(name.stripBackticks)
-  def collect(tree: Tree, toAdjust: Position): TextEdit = {
+  private val forbiddenMethods =
+    Set("equals", "hashCode", "unapply", "unary_!", "!")
+
+  def canRenameSymbol(sym: Symbol): Boolean = {
+    (!sym.isMethod || !forbiddenMethods(sym.decodedName)) &&
+    (sym.ownersIterator
+      .drop(1)
+      .exists(
+        _.isMethod
```

Comment:
We could have that only for the actual rename and just have general check for both versions here.